### ISSUE TITLE
support for count in queries

### ISSUE THREAD

**Issue by abstratt** [abstratt/cloudfier#71]
```
    static query pendingExpenseCount() : Integer;
    begin
    	return Expense extent.count((e : Expense) : Boolean {
            e.status == Status#Submitted
        });
    end;
```

```
public long pendingExpenseCount() {
        CriteriaBuilder cb = getEntityManager().getCriteriaBuilder();
        CriteriaQuery<Long> cq = cb.createQuery(Long.class);
        Root<Expense> expense_ = cq.from(Expense.class);
        return getEntityManager().createQuery(
            cq.distinct(true).where(
                cb.equal(expense_.get("status"), Expense.Status.Submitted)
    		).select(cb.count(expense_))
        ).getSingleResult();
    }
```

**Comment by abstratt** [comment]
See also #70.

**Comment by abstratt** [comment]
Fixed in 4e9b9ff3af946ed207188599eb1170ec59579fce.
